File: frontend/src/types/mediaKit.ts
export interface MediaKitTemplate {
  id: number
  name: string
  style: 'MODERN' | 'CLASSIC' | 'MINIMAL' | 'CREATIVE'
  previewUrl?: string
}

export interface AudienceDemographic {
  ageGroup: string
  percentage: number
  gender: 'MALE' | 'FEMALE' | 'OTHER'
}

export interface PlatformStat {
  platform: string
  followers: number
  avgViews: number
  engagementRate: number
  growthRate: number
}

export interface CampaignResult {
  id: number
  brandName: string
  campaignName: string
  platform: string
  views: number
  engagement: number
  roi: number
  date: string
}

export interface RateCard {
  type: 'SPONSORED_VIDEO' | 'PRODUCT_PLACEMENT' | 'STORY_POST' | 'SHORT_FORM' | 'LIVE_STREAM' | 'BUNDLE'
  price: number
  currency: string
  description: string
}

export interface MediaKit {
  id: number
  title: string
  templateStyle: MediaKitTemplate['style']
  bio: string
  profileImageUrl?: string
  platforms: PlatformStat[]
  demographics: AudienceDemographic[]
  topContent: { title: string; views: number; thumbnail?: string }[]
  campaignResults: CampaignResult[]
  rateCards: RateCard[]
  contactEmail: string
  createdAt: string
  updatedAt: string
  publishedUrl?: string
  status: 'DRAFT' | 'PUBLISHED'
}

export interface MediaKitGenerateRequest {
  templateStyle: MediaKitTemplate['style']
  includeRateCards: boolean
  includeCampaigns: boolean
  customBio?: string
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/mediakit/MediaKit.kt
package com.ongo.domain.mediakit

import java.time.LocalDateTime

data class MediaKit(
    val id: Long? = null,
    val userId: Long,
    val title: String,
    val templateStyle: String = "MODERN",
    val bio: String? = null,
    val profileImageUrl: String? = null,
    val platforms: String = "[]",
    val demographics: String = "[]",
    val topContent: String = "[]",
    val campaignResults: String = "[]",
    val rateCards: String = "[]",
    val contactEmail: String? = null,
    val status: String = "DRAFT",
    val publishedUrl: String? = null,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
)
